File: api/src/endpoint/oauth/authorize_device.rs
use api_derive::ApiRequest;
use serde::{Deserialize, Serialize};

use crate::endpoint::Endpoint;

pub struct AuthorizeDevice;

impl Endpoint for AuthorizeDevice {
    const PATH: &'static str = "/oauth/authorize";
    const METHOD: http::Method = http::Method::POST;

    type Request = AuthorizeDeviceRequest;
    type Response = ();
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct AuthorizeDeviceRequest {
    pub user_code: String,
}

File: gitdot-api-ts/src/endpoint/oauth/authorize-device.ts
import { z } from "zod";

export const AuthorizeDeviceRequest = z.object({
  user_code: z.string(),
});
export type AuthorizeDeviceRequest = z.infer<typeof AuthorizeDeviceRequest>;

export const AuthorizeDevice = {
  path: "/oauth/authorize",
  method: "POST",
  request: AuthorizeDeviceRequest,
  response: z.void(),
} as const;
export type AuthorizeDevice = typeof AuthorizeDevice;
